src/webgpu/shader/validation/decl/var_access_mode.spec.ts

```diff
@@ -0,0 +1,226 @@
+export const description = `
+7.3 var Declarations
+
+The access mode always has a default value, and except for variables in the
+storage address space, must not be specified in the WGSL source. See §13.3 Address Spaces.
+`;
+
+// TODO(4128): Validate the similar rules on pointer types.
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import {
+  AccessMode,
+  AddressSpace,
+  AddressSpaceInfo,
+  kAccessModeInfo,
+  kAddressSpaceInfo,
+} from '../../types.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+import { declareEntryPoint, ShaderStage } from './util.js';
+
+// Test address spaces that we can hold an i32 variable.
+const kOrdinaryAddressSpaces = keysOf(kAddressSpaceInfo).filter(as => as !== 'handle');
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+/**
+ * @returns a WGSL declaration for an 'x' variable with the given parameters.
```

Comment:
Tint uses `variable` as the general collective term for `var`, `let`, `override` `const` and parameters.

```suggestion
 * @returns a WGSL snippet that declares a module-scope 'var' with name 'x' using the given parameters.
```